Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/use-contests.tsx

```diff
@@ -25,41 +28,58 @@ interface IContestsProviderProps extends IHaveChildrenProps {
 
 const defaultState = {
     state: {
-        contests: [] as IContestType[],
+        contests: [] as IIndexContestsType[],
         possibleFilters: [
-            ...generateFilterItems(FilterType.Status, ContestState.Active, ContestState.Past),
-            ...generateFilterItems(FilterType.Language, 'Py', 'JS', 'Java', 'C#', 'Go'),
+            ...generateFilterItems(
+                FilterType.Status,
+                { name: ContestState.Active, value: ContestState.Active },
+                { name: ContestState.Past, value: ContestState.Past }),
         ],
         filters: [] as IFilter[],
     },
+    pageNumber: 1,
 };
 
 const ContestsContext = createContext<IContestsContext>(defaultState as IContestsContext);
 
 const ContestsProvider = ({ children }: IContestsProviderProps) => {
     const [ contests, setContests ] = useState(defaultState.state.contests);
-    const [ possibleFilters ] = useState(defaultState.state.possibleFilters);
+    const [ possibleFilters, setPossibleFilters ] = useState(defaultState.state.possibleFilters);
     const [ filters, setFilters ] = useState(defaultState.state.filters);
+    const [ page, setPage ] = useState(defaultState.pageNumber);
+    const [ pageProps, setPageProps ] = useState({} as IHavePagesProps);
 
     const { getUrlForAllContests } = useUrls();
+    const { startLoading, stopLoading } = useLoading();
 
     const getUrl = useCallback(
-        () => getUrlForAllContests({ filters }),
-        [ filters, getUrlForAllContests ],
+        () => getUrlForAllContests({ filters, page }),
+        [ filters, getUrlForAllContests, page ],
     );
 
     const {
         get,
         data,
     } = useHttp(getUrl);
+    
+    const {
+        state: {
+            strategies,
+        },
+    } = useContestStrategyFilters();
 
     const applyFilter = useCallback(
-        (filter) => {
-            const newFilters = filters.indexOf(filter) >= 0
-                ? without(filters, filter)
-                : [ ...filters, filter ];
+        (filter: IFilter, singleForType = false) => {
+            let newFilters = isNil(filters.find(f => f.id === filter.id))
```

Comment:
As in previous code, `some` can be used here, i.e. `filters.some(({id}) => id === filter.id)`